pandas/io/parquet.py

```diff
@@ -147,27 +148,30 @@ def to_parquet(df, path, engine='auto', compression='snappy', **kwargs):
 
     valid_types = {'string', 'unicode'}
 
-    # validate index
-    # --------------
-
     # validate that we have only a default index
     # raise on anything else as we don't serialize the index
-
-    if not isinstance(df.index, Int64Index):
-        raise ValueError("parquet does not support serializing {} "
-                         "for the index; you can .reset_index()"
-                         "to make the index into column(s)".format(
-                             type(df.index)))
-
-    if not df.index.equals(RangeIndex.from_range(range(len(df)))):
-        raise ValueError("parquet does not support serializing a "
-                         "non-default index for the index; you "
-                         "can .reset_index() to make the index "
-                         "into column(s)")
-
-    if df.index.name is not None:
-        raise ValueError("parquet does not serialize index meta-data on a "
-                         "default index")
+    # *unless* we're using pyarrow >= 0.7.1 which does support multi-indexes
+    if impl.api.__name__ == 'pyarrow' and not impl._pyarrow_lt_070:
+        validate_index = False
+    else:
+        validate_index = True
+
+    if validate_index:
+        if not isinstance(df.index, Int64Index):
+            raise ValueError("parquet does not support serializing {} "
+                             "for the index; you can .reset_index()"
+                             "to make the index into column(s)".format(
+                                 type(df.index)))
+
+        if not df.index.equals(RangeIndex.from_range(range(len(df)))):
```

Comment:
Could create the `RangeIndex` via `RangeIndex(len(df))` here to make this a little cleaner.

Reply:
Done.